### DISCUSSION TITLE
runtime-attach not working, I'm running on my main.

### DISCUSSION THREAD

**Discussion by emsilva** [open-telemetry/opentelemetry-java-instrumentation#9911] [Q&A]
I'm trying to use this fature: [runtime-attach](https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/runtime-attach).
I'm getting the error: "Agent is not attached because runtime attachment was not requested from main thread."

As you can see, it is on my main. This is a flink job and I'm managing it through Maven. Any changes I should do or known workarounds?

```
import io.opentelemetry.contrib.attach.RuntimeAttach;
import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaProducer;
import org.apache.flink.configuration.Configuration;

import java.util.Properties;

public class KafkaToFlinkToKafka {
    public static void main(String[] args) throws Exception {
        RuntimeAttach.attachJavaagentToCurrentJvm();
...
    }
}
```

my pom:

```
<dependency>
	<groupId>io.opentelemetry.contrib</groupId>
	<artifactId>opentelemetry-runtime-attach</artifactId>
	<version>1.31.0-alpha</version>
</dependency>
```

I'm using Java 11, Flink 1.17. for reference.

**Comment by laurit** [comment]
That check verifies that you are doing the runtime attach right after the jvm has started. If you are running your application from maven this would only be the case if the execution is forked.
